var/spack/repos/builtin/packages/cppad/package.py

```diff
@@ -9,20 +9,32 @@
 class Cppad(CMakePackage):
     """A Package for Differentiation of C++ Algorithms."""
 
+    maintainers("whart222")
+
     homepage = "https://www.coin-or.org/CppAD/"
-    url = "http://www.coin-or.org/download/source/CppAD/cppad-20170114.gpl.tgz"
+    url = "https://github.com/coin-or/CppAD/archive/refs/tags/20240000.4.tar.gz"
     git = "https://github.com/coin-or/CppAD.git"
 
     version("develop", branch="master")
+    version(
+        "20240000.4", sha256="0dfc1e30b32d5dd3086ee3adb6d2746a019e9d670b644c4d5ec1df3c35dd1fe5"
+    )
+    version(
+        "20220000.5", sha256="9fb4562f6169855eadcd86ac4671593d1c0edf97bb6ce7cbb28e19af2bfc165e"
+    )
     version(
         "20180000.0", sha256="1c355713e720fc5226ff3d6db2909922d46cd7ee0d36ee7985882f86905f655a"
     )
     version("20170114", sha256="fa3980a882be2a668a7522146273a1b4f1d8dabe66ad4aafa8964c8c1fd6f957")
```

Comment:
I cannot confirm the sha256 for this version, not with the new URL or the old one.  Looks like it isn't in the GitHub repo.

Perhaps it should be deprecated?  See https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions.

Reply:
I deprecated both of the two old versions. Do you think I should update their sha256 values based on the current github tarballs? I don't know who setup these versions initially.